### PR TITLE
[20542] Fix CI version management

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR unifies the CI in all workflows to follow this behavior:

- Trigger CI in all pull requests which HEAD branch is set to this supported branch version
- Remove any CI trigger on push events and nightly
- Include missing ignore files for the sanitizer event trigger.
- Rename sanitizer job as "address-sanitizer"

master related PR:
- #4480 

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Changes are ABI compatible. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: .github/workflows/sanitizer-tests.yaml
Comment (elianalf) [inline .github/workflows/sanitizer-tests.yaml:1]: [Line 9](https://github.com/eProsima/Fast-DDS/pull/4482/files#diff-37fd8971cb880bd169034bc59986b197ec187bd7193ef6edfd92a85d5e40ba07R9) and [line 132](https://github.com/eProsima/Fast-DDS/pull/4482/files#diff-37fd8971cb880bd169034bc59986b197ec187bd7193ef6edfd92a85d5e40ba07R132) should be `2.12.x`

**Review by elianalf — CHANGES_REQUESTED** [review]

File: .github/workflows/sanitizer-tests.yaml
Comment (JesusPoderoso) [reply .github/workflows/sanitizer-tests.yaml:1]: I think so, yes. Lets discuss it with the team

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Review by elianalf — DISMISSED** [review]

**Review by elianalf — APPROVED** [review]

**Comment by elianalf** [comment]
@richiprosima please test linux

**Review by elianalf — DISMISSED** [review]

**Comment by JesusPoderoso** [comment]
Documentation Manual CI run [here](https://github.com/eProsima/Fast-DDS/actions/runs/8272706465)
Consider Ready to merge if passes, it is depending on this documentation PR:
- eProsima/Fast-DDS-docs#727

**Pull request merged**